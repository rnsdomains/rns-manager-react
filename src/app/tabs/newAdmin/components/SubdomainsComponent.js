import React from 'react';
import propTypes from 'prop-types';
import { multilanguage } from 'redux-multilanguage';

const SubdomainsComponent = (props) => {
  const { strings } = props;
  return (
    <div>
      <h1>{strings.subdomains}</h1>
    </div>
  );
};

SubdomainsComponent.propTypes = {
  strings: propTypes.shape({
    subdomains: propTypes.string.isRequired,
  }).isRequired,
};

export default multilanguage(SubdomainsComponent);
