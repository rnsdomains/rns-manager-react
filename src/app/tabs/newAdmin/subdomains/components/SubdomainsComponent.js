import React from 'react';
// import propTypes from 'prop-types';
// import { multilanguage } from 'redux-multilanguage';

import { NewSubdomainContainer, SubdomainListContainer } from '../containers';

const SubdomainsComponent = () => (
  <div className="subdomains">
    <NewSubdomainContainer />
    <SubdomainListContainer />
  </div>
);

/*
SubdomainsComponent.propTypes = {
  strings: propTypes.shape({
  }).isRequired,
};
*/

export default SubdomainsComponent;
// export default multilanguage(SubdomainsComponent);
