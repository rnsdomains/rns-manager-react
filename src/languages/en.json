{
  "home_title": "rif name service",
  "home_subtitle": "get your domain, forget your address",
  "register_your_domain": "register your domain",
  "what_is_rns_title": "what is rns?",
  "what_is_rns_text": "RIF Name Service provides an architecture which enables the identification of blockchain addresses by human-readable names.",
  "resolve_a_domain": "resolve a domain",
  "admin_your_domain_title": "admin your domain",
  "admin_your_domain_action_1": "admin domain",
  "admin_your_domain_action_2": "admin domain reolution",
  "admin_your_domain_action_3": "create subdomains",
  "admin": "admin",
  "develop": "develop",
  "libs": "libs",
  "docs": "docs",
  "github": "github",
  "learn": "learn",
  "rif_white_paper": "rif white paper",
  "rns_white_paper": "rns white paper",
  "search": "search",
  "resolve": "resolve",
  "invalid_name": "Invalid name. Must be lower case characters and/or numbers",
  "open": "open",
  "auction": "auction",
  "finalize": "finalize",
  "reveal": "reveal",
  "owned": "owned",
  "search_another_domain": "search another domain",
  "bid_in_auction": "bid in the domain auction",
  "finalize_the_auction": "finalize the auction",
  "reveal_your_bid": "reveal your bid",
  "learn_about_process": "learn about process",
  "hide": "hide",
  "process": "Any user can register any available domain name. Registering a name requires to complete 3 steps",
  "process_step_1": "Request to register",
  "process_step_1_explanation": "Your wallet will open and you will be asked to confirm the first of two transactions required for registration.",
  "process_step_2": "Wait for a minute",
  "process_step_3": "Register",
  "process_step_3_explanation": "Click ‘register’ and your wallet will re-open. Only after the 2nd transaction is confirmed you'll know if you got the domain",
  "learn_more": "learn more",
  "mycrypto_search_title": "Search status on MyCrypto",
  "mycrypto_select_network": "Select RSK network on the top right selector.",
  "mycrypto_go_to_interact": "Go to My Crypto contract interaction tab.",
  "mycrypto_select_registrar": "Select RNS Registrar contract on Existing Contract selector.",
  "mycrypto_access": "Access!",
  "mycrypto_on_read_write_select": "On Read/Write Contract select",
  "my_crypto_copy_paste_on": "Copy and paste this hash on",
  "mycrypto_read": "Read!",
  "result": "Result",
  "state": "State",
  "next_action": "Next action",
  "go_to_mycrypto": "Go to MyCrypto",
  "start_an_auction_for": "start an auction for",
  "start_auction": "start auction",
  "mycrypto_start_title": "Start auction in MyCrypto",
  "mycrypto_choose_checkout": "Choose your checkout method.",
  "mycrypto_check_gas": "Check gas according to",
  "mycrypto_write": "Write!",
  "bid_for": "bid for",
  "how_much_do_you_value": "how much do you value",
  "bid_error_one_rif": "You must bid at least 1 RIF",
  "secret_phrase": "secret phrase",
  "random": "random",
  "advanced_options": "advanced options",
  "bid": "bid",
  "mycrypto_bid_title": "Bid in MyCrypto",
  "mycrypto_seal_the_bid": "First we are going to seal the bid",
  "mycrypto_copy_paste_values": "Copy this values and paste them in MyCrypto fields",
  "mycrypto_bid_select_owner": "the domain's owner. You may want to use your wallet address.",
  "mycrypto_remember_to_save_bid_data": "Remember you must save all this data to unseal the bid on reveal phase!",
  "mycrypto_copy_response": "Copy the response here",
  "mycrypto_select_rif": "Select RIF contract on Existing Contract selector.",
  "finalize_auction_for": "finalize auction for",
  "finalize_auction": "finalize auction",
  "mycrypto_finalize_title": "Finalize auction on MyCrypto",
  "unseal_bid_for": "unseal your bid for",
  "upload_bid_data": "upload bid data",
  "value": "value",
  "unseal_bid": "unseal bid",
  "mycrypto_unseal_title": "Unseal bid on MyCrypto",
  "mycrypto_admin_title": "admin your domain in MyCrypto",
  "name": "name",
  "admin_ownership": "admin ownership",
  "field": "field",
  "choose_": "choose...",
  "owner": "owner",
  "resolver": "resolver",
  "ttl": "tll",
  "get": "get",
  "set": "set",
  "admin_subdomain_ownership": "admin subdomain ownership",
  "subdomain": "subdomain",
  "admin_resolution": "admin resolution",
  "check_domain_resolver": "check domain resolver",
  "resolver_result": "resolver result",
  "public_resolver": "Public Resolver",
  "addr": "addr",
  "content": "content",
  "rsk_wallet_needed": "you need an RSK wallet to resolve, register and admin domains",
  "get_metamask": "get Metamask",
  "use_with_mycrypto": "use with MyCrypto",
  "connect_metamask_to_rsk" : "connect Metamask to RSK network",
  "connect": "connect",
  "done": "done",
  "subdomains": "subdomains",
  "Get": "Get",
  "on_mycrypto": "on MyCrypto",
  "mycrypto_select_registry": "Select RNS Registry contract on Existing Contract selector.",
  "Set": "Set",
  "mycrypto_select_resolver": "Select RNS Resolver contract on Existing Contract selector.",
  "mycrypto_select_multi_resolver": "Select Multi-crypto Resolver contract on Existing Contract selector.",
  "log_in": "log in",
  "address": "address",
  "network": "network",
  "not_domains_owner_message": "You are not the domains's owner.",
  "get_the_domain": "Get the domain!",
  "switch_to_mycrypto": "switch to MyCrypto",
  "register": "register",
  "please_log_in": "Please log in.",
  "switch_to_metamask": "switch to Metamask",
  "start": "start",
  "edit": "edit",
  "cancel": "cancel",
  "notifications": "notifications",
  "unread_notifications": "unread notifications",
  "Confirmed": "confirmed",
  "Waiting_for_confirmation_": "waiting for confirmation...",
  "set_owner": "set owner",
  "language": "language",
  "active_wallet": "active wallet",
  "setup_title": "Set up",
  "setup_download_metamask": "Download Metamask",
  "setup_connect_to_rsk": "Connect Metamask to RSK network:",
  "setup_open_network_selector": "Open network selector",
  "setup_select_custom_rpc": "Select 'Custom RPC'",
  "setup_select_advanced": "In new RPC url select 'Advanced options' and fill",
  "setup_save": "Save",
  "setup_add_rif": "Add RIF Token view (optional)",
  "setup_open_left_menu": "Open left menu",
  "setup_go_to_add_token": "Go to 'Add token'",
  "setup_fill_token_address_with_": "In custom token tab fill Token address with: ",
  "setup_next": "Next",
  "setup_add_tokens": "Add tokens",
  "notifications_start_auction_title": "auction started",
  "notifications_start_auction_action": "make your bid",
  "notifications_bid_title": "new bid",
  "notifications_bid_dont_forget": "don't forget to unseal the bid!",
  "notifications_bid_event_title": "Unseal bid for",
  "notifications_bid_download_message": "you will need the bid data to unseal it.",
  "download": "download data",
  "notifications_unseal_title": "Unsealed bid",
  "notifications_unseal_dont_forget": "don't forget to finelize the auction!",
  "notifications_unseal_event_title": "Finalize auction for",
  "notifications_finalize_title": "auction finalized",
  "notifications_finalize_action": "set addr resolution",
  "notifications_new_owner": "new owner",
  "notifications_new_resolver": "new resolver",
  "notifications_new_ttl": "new ttl",
  "notifications_new_subdomain_owner": "new subdomain owner",
  "notifications_new_addr": "new addr",
  "notifications_new_content": "new content",
  "rsk_address": "rsk address",
  "chain_address": "chain address",
  "bitcoin": "bitcoin (BTC)",
  "ethereum": "ethereum (ETH)",
  "bitcoin_cash": "bitcoin cash (BCH)",
  "litecoin": "litecoin (LTC)",
  "log_out": "log out",
  "notifications_new_chain_addr": "new chain addr",
  "rsk": "rsk (RBTC)",
  "suggested_subdomains": "wallet, donate, me, ...",
  "rsk_resolver": "rsk address resolver",
  "multi_chain_resolver": "multi crypto address resolver",
  "migrate_multichain_resolver_title": "New feature available!",
  "migrate_multichain_resolver_message": "Migrate to the new Multi Chain Resolver. It provides your domain of addresses for all assets, for example, your Bitcoin address.",
  "set_address": "set address",
  "setup_fill_with": "Fill with:",
  "select_chain": "select chain",
  "get_value_on_mycrypto": "get value on MyCrypto",
  "get_owner": "get owner",
  "reverse_resolution": "reverse resolution",
  "set_reverse": "set reverse",
  "has_reverse": "already set",
  "notifications_new_reverse_resolution": "new reverse resolution registered",
  "fifs_migration": "fifs registrar migration",
  "already_migrated": "your domain has been succesfully migrated to the fifs registrar",
  "fifs_subdomain": "subdomains don't need to be migrated",
  "fifs_migrate": "migrate to fifs registrar",
  "due_migration_date": "you have to migrate your domain to the new FIFS registrar by clicking on the button below. the due migration date is DD/MM/YYYY 12:00 AM UTC",
  "action_needed": "action needed",
  "notifications_migrated_domain": "your domain has been migrated to the new fifs registrar",
  "domain_not_available": "not available",
  "start_registration_for": "start the registration for",
  "rental_period": "rental period in years",
  "commit": "request for registration",
  "notifications_registrar_committed": "your domain has been requested to be registered",
  "notifications_registrar_revealed":  "your domain has been registered",
  "login_explanation": "now you can login with your just registered domain",
  "rskTld": ".rsk",
  "blocked_domain": "domains with less than 5 characters are blocked. stay tuned, we are going to release them soon",
  "transfer_domain": "transfer domain",
  "transfer": "transfer",
  "already_transferred": "your domain ownership has been succesfully transferred",
  "notifications_transferred_domain_token": "your domain ownership has been succesfully transferred",
  "renew_your_domain": "renew your domain",
  "your_domain_expires": "Your domain expires in ",
  "days": " days.",
  "not_registered_domain": "The domain has not been registered or has expired, you should register it before renewing.",
  "start_renovation_for": "start renovation for",
  "renew": "renew",
  "renew_domain_successfully": "the domain has been renewed succesfully",
  "discount": "50% discount per year from the third year",
  "rif_per_year": "RIF per year",
  "price": "price",
  "how_long_want_name": "For how long do you want your name",
  "not_enough_balance": "you don't have enough balance to perform this action",
  "click_here_not_enough_balance": "click here and check where you can buy more tokens",
  "no_wallet": "No wallet detected",
  "network_mismatch": "Network Mismatch",
  "connect_to_network": "Connect your wallet to",
  "current_connected": "Your wallet is currently connected to",
  "connected_successful": "Connected Successfully",
  "unknown_network": "Unknown Network",
  "unlock_wallet": "Please unlock your wallet",
  "string_resolver": "String resolver",
  "notifications_new_str": "new str",
  "admin_str_resolution": "admin str resolution",
  "address_formatted_incorrectly" : "Your address is not formatted for this network.",
  "address_formatted_incorrectly_description" : "Please check that the address is correct before continuing. If the address is correct, please use the button below to convert your address to lowercase and proceed.",
  "convert_to_lowercase": "Convert address to lowercase and continue.",
  "registering": "Registering",
  "request_domain": "Request domain",
  "register_domain": "Register domain",
  "login": "Login",
  "did_you_know": "Did you know that",
  "tip1": "You can use your domain in RIF Marketplace to provide your services.",
  "read_more": "Read more about it",
  "domain_requested": "Your domain has been requested",
  "to_register": "To finish registering",
  "admin_domain": "Admin my domain",
  "register_another_domain": "Register another domain",
  "wait_for_two_minutes": "Wait for aproximately two minutes",
  "wait_period_reason": "the waiting period is required to securely request your domain.",
  "click_register_domain": "click 'Register domain' and your wallet will re-open.",
  "your_domain_will_be_registered": "Your domain will be registered after the transaction is confirmed.",
  "wait_transation_confirmed": "Wait until the transaction is confirmed. It will take less than a minute.",
  "how_long_want_domain": "How long do you want your domain",
  "period_in_years": "period in years",
  "auto_address_setup": "Automatic address setup",
  "auto_address_explanation": "If you choose this option, your wallet address will automatically be set as the resolved address of your domain.",
  "notification_domain_ready_register": "Your domain is ready to be registered",
  "advanced": "Advanced",
  "basic": "Basic",
  "domain_info": "Domain info",
  "your_addresses": "Your addresses",
  "reverse": "Reverse",
  "copied": "copied",
  "copy_text": "Copy text",
  "transfer_warning": "Transfer the ownership of your domain to another address. Please be aware that once you transfer the ownership of your domain you will not be able to access and manage the domain anymore.",
  "expires_on": "Expires on",
  "renew_explanation": "Your wallet will open and you will be asked to confirm the first of two transactions required for renovation.",
  "renew_domain": "Renew Domain",
  "close": "Close",
  "renew_success": "Your domain has been renewed.",
  "success": "Success",
  "view_explorer": "View it on the explorer",
  "was_transfered": "was transfered",
  "register_new_domain": "Register a new domain",
  "login_another_domain": "Login to another domain",
  "upgrade_domain": "Upgrade domain",
  "upgrade": "upgrade",
  "upgrade_domain_explanation": "The domain upgrade offers new functionalities and dynamic renovation. The upgrade requires a single transaction with your wallet, then your domain will be upgraded automatically.",
  "faq": "Frequently Asked Questions"
}
