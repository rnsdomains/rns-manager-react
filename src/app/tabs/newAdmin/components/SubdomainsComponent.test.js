import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import mockStore from '../../../../../tests/config/mockStore';
import en from '../../../../languages/en.json';

import SubdomainsComponent from './SubdomainsComponent';

const store = mockStore({
  subdomains: en.subdomains,
});

const component = mount(<Provider store={store}><SubdomainsComponent /></Provider>);

describe('DomainInfoComponent', () => {
  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
