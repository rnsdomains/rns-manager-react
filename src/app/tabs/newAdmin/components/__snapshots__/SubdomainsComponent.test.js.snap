// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainInfoComponent matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(SubdomainsComponent)>
    <SubdomainsComponent
      currentLanguageCode="en"
      dispatch={[Function]}
      strings={
        Object {
          "subdomains": "subdomains",
        }
      }
    >
      <div>
        <h1>
          subdomains
        </h1>
      </div>
    </SubdomainsComponent>
  </Connect(SubdomainsComponent)>
</Provider>
`;
